fdbserver/include/fdbserver/workloads/MetaclusterConsistency.actor.h

```diff
@@ -71,55 +74,69 @@ class MetaclusterConsistencyCheck {
 	ACTOR static Future<Void> loadManagementClusterMetadata(MetaclusterConsistencyCheck* self) {
 		state Reference<typename DB::TransactionT> managementTr = self->managementDb->createTransaction();
 		state KeyBackedRangeResult<std::pair<int64_t, TenantMapEntry>> tenantList;
+		state Optional<int64_t> lastTenantId;
 
 		loop {
 			try {
 				managementTr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
 				state typename transaction_future_type<typename DB::TransactionT, RangeResult>::type
-				    systemTenantSubspaceKeysFuture = managementTr->getRange(prefixRange(TenantMetadata::subspace()), 1);
-
-				wait(store(self->managementMetadata.metaclusterRegistration,
-				           MetaclusterMetadata::metaclusterRegistration().get(managementTr)) &&
-				     store(self->managementMetadata.dataClusters,
-				           MetaclusterAPI::listClustersTransaction(
-				               managementTr, ""_sr, "\xff\xff"_sr, CLIENT_KNOBS->MAX_DATA_CLUSTERS + 1)) &&
-				     store(self->managementMetadata.clusterCapacityTuples,
-				           MetaclusterAPI::ManagementClusterMetadata::clusterCapacityIndex.getRange(
-				               managementTr, {}, {}, CLIENT_KNOBS->MAX_DATA_CLUSTERS)) &&
-				     store(self->managementMetadata.clusterTenantCounts,
-				           MetaclusterAPI::ManagementClusterMetadata::clusterTenantCount.getRange(
-				               managementTr, {}, {}, CLIENT_KNOBS->MAX_DATA_CLUSTERS)) &&
-				     store(self->managementMetadata.clusterTenantTuples,
-				           MetaclusterAPI::ManagementClusterMetadata::clusterTenantIndex.getRange(
-				               managementTr, {}, {}, metaclusterMaxTenants)) &&
-				     store(self->managementMetadata.clusterTenantGroupTuples,
-				           MetaclusterAPI::ManagementClusterMetadata::clusterTenantGroupIndex.getRange(
-				               managementTr, {}, {}, metaclusterMaxTenants)) &&
-				     store(self->managementMetadata.tenantCount,
-				           MetaclusterAPI::ManagementClusterMetadata::tenantMetadata().tenantCount.getD(
-				               managementTr, Snapshot::False, 0)) &&
-				     store(tenantList,
-				           MetaclusterAPI::ManagementClusterMetadata::tenantMetadata().tenantMap.getRange(
-				               managementTr, {}, {}, metaclusterMaxTenants)) &&
-				     store(self->managementMetadata.tenantGroups,
-				           MetaclusterAPI::ManagementClusterMetadata::tenantMetadata().tenantGroupMap.getRange(
-				               managementTr, {}, {}, metaclusterMaxTenants)) &&
-				     store(self->managementMetadata.systemTenantSubspaceKeys,
-				           safeThreadFutureToFuture(systemTenantSubspaceKeysFuture)));
+				    systemTenantSubspaceKeysFuture = managementTr->getRange(prefixRange(TenantMetadata::subspace()), 2);
+				state Optional<int64_t> tenantIdPrefix = wait(TenantMetadata::tenantIdPrefix().get(managementTr));
```

Comment:
This function was written to load all of the metadata concurrently in the wait block below. Then the various checks (like the new assertions) are mostly placed in `validateManagementCluster`. Can you update the new metadata to follow a similar pattern?

Reply:
sure